sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/EventPropertiesTest.java

```diff
@@ -0,0 +1,208 @@
+package com.microsoft.appcenter.analytics;
+
+import com.microsoft.appcenter.ingestion.models.properties.BooleanTypedProperty;
+import com.microsoft.appcenter.ingestion.models.properties.DateTimeTypedProperty;
+import com.microsoft.appcenter.ingestion.models.properties.DoubleTypedProperty;
+import com.microsoft.appcenter.ingestion.models.properties.LongTypedProperty;
+import com.microsoft.appcenter.ingestion.models.properties.StringTypedProperty;
+import com.microsoft.appcenter.utils.AppCenterLog;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.rule.PowerMockRule;
+
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+
+@PrepareForTest({
+        AppCenterLog.class
+})
+public class EventPropertiesTest {
+
+    @Rule
+    public PowerMockRule mPowerMockRule = new PowerMockRule();
+
+    @Before
+    public void setUp() {
+        mockStatic(AppCenterLog.class);
+    }
+
+    @Test
+    public void keyValidate() {
```

Comment:
This is also an overloaded test... Less egregious than the last one but it's still doing a lot.  We can just make this easier on ourselves by splitting it.

Reply:
Split in d51458a